server/lib/accounts.js

```diff
@@ -168,6 +168,12 @@ Accounts.validateLoginAttempt(function(login) {
 		});
 	}
 
+	if (!{}.hasOwnProperty.call(login.user, 'roles') || !Array.isArray(login.user.roles)) {
```

Comment:
I'm not sure about others, but I find `!{}.hasOwnProperty.call` to be a tad confusing at first glance. Can we change this to something like `if (login.user && (!login.user.roles || Array.isArray(login.user.roles)) {}`?

Reply:
Why not?! I will change and update the branch too.

Sorry about that, I like to follow [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins).

Reply:
Because if you look through the rest of our code base, you will see the style is how I mentioned. So, it's more about being consistent than anything to be honest.

Reply:
Maybe we should look at making part of our style guide if this is a more proper way of checking for the property?

But I do agree on we need to stay consistent, I don't think we use that elsewhere.  Not to say we shouldn't... just we don't so does make it stick out.

Even a comment with a note about the style guide would help a bit in regards to readability here.

All for better code.

Reply:
I agree about the consistency of style. In this particular piece of code, even the user.roles is undef or null or false, we have the second criteria to check if is array, so this dont cause any bug.

But is good idea too make the things in most safety away. Is a good idea add this in style guide in future.

Reply:
I believe we have an issue discussion code style #1163 this would be good to add to discussion as something to include

Reply:
Sure @geekgonecrazy :)